File: Assignment-2/Assn2.cpp
//author: Sotheanith Sok

//Import
#include<iostream>
#include<string>
#include<fstream>
#include <regex>

using namespace std;

int main(int argc, char **argv) {
    string line;
    //Define regex
    regex e("^(\\$)(\\**)(([0-9])|(([1-9][0-9]?[0-9]?)((,)[0-9][0-9][0-9])*))((((\\.)[0-9][0-9])*)?)$");

    //Create reader
    ifstream out(argv[1]);

    //Read, compare, and print out result
    while(getline(out, line)) {
       if(regex_match(line,e)){
         cout<<"Matched: "<<line<<endl;
       }else{
          cout<<"Not Matched: "<<line<<endl;
       }
    }
    
    //Close reader
    out.close();
    return 0;
}

File: Assignment-2/Assn2.js
//author: Sotheanith Sok
//Imports
const fs = require('fs');
const readline=require('readline');

//Define regex
const regex = new RegExp('^\\$\\**(0|(([1-9][0-9]?[0-9]?)(,[0-9][0-9][0-9])*))((\\.[0-9][0-9])*)?$');

//Create read stream
const rl = readline.createInterface({
  input: fs.createReadStream('./'+process.argv[2]),
  crlfDelay: Infinity
});

//Compare and print out result
var result;
rl.on('line', (line) => {
     result = regex.test(line);
    if(result){
       console.log('Matched: '+line);
    }else{
       console.log('Not Matched: '+line);
    }
});